pyccel/ast/datatypes.py

```diff
@@ -12,115 +12,201 @@
 
 import numpy
 
-from pyccel.utilities.metaclasses import ArgumentSingleton, Singleton
+from pyccel.utilities.metaclasses import Singleton, ArgumentSingleton
 
-# TODO [YG, 12.03.2020] verify why we need all these types
-# NOTE: symbols not used in pyccel are commented out
 __all__ = (
-#
-# --------- CLASSES -----------
-#
-    'CustomDataType',
-    'DataType',
-    'DataTypeFactory',
-    'NativeBool',
-    'NativeComplex',
-    'NativeFloat',
-    'NativeGeneric',
-    'NativeHomogeneousList',
-    'NativeHomogeneousSet',
-    'NativeHomogeneousTuple',
-    'NativeInhomogeneousTuple',
-    'NativeInteger',
-    'NativeString',
-    'NativeSymbol',
-    'NativeTuple',
-    'NativeVoid',
-#
-# --------- FUNCTIONS -----------
-#
-    'datatype',
-    'str_dtype',
-#
-# --------- VARIABLES -----------
-#
-    'Bool',
-    'Cmplx',
-    'Generic',
-    'Int',
-    'Float',
-    'String',
-    'Void',
-	'NativeNumeric',
-#    '_Symbol',
-    'default_precision',
-    'dtype_and_precision_registry',
+        # ------------ Super classes ------------
+        'PyccelType',
+        'PrimitiveType',
+        'FixedSizeType',
+        'ContainerType',
+        # ------------ Primitive types ------------
+        'PyccelBooleanType',
+        'PyccelIntegerType',
+        'PyccelFloatingPointType',
+        'PyccelComplexType',
+        'PyccelCharacterType',
+        # ------------ Fixed size types ------------
+        'FixedSizeNumericType',
+        'PythonNativeNumericTypes',
```

Comment:
Why using the plural "Types" for this class name?

Reply:
It used to be a tuple of types rather than a superclass which is why it ended up with this name. Fixed : 5f92c84d7